firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionLifecycleService.kt

```diff
@@ -89,12 +107,14 @@ internal class SessionLifecycleService : Service() {
   private fun handleForegrounding(msg: Message) {
     Log.i(TAG, "Activity foregrounding at ${msg.getWhen()}")
 
-    if (curSessionId == null) {
+    if (!hasForegrounded) {
       Log.i(TAG, "Cold start detected.")
-      newSession()
+      hasForegrounded = true
+      broadcastSession()
```

Comment:
Don't we need to generate a new session here too?

Reply:
The sessionGenerator creates an initial session right away, so on cold start we need to use that one instead of generating a new one: https://github.com/firebase/firebase-android-sdk/blob/42ec418ad57a40230636481937fca541b0c6c8ad/firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionGenerator.kt#L42